x-pack/plugins/alerting/server/task_runner/create_execution_handler.ts

```diff
@@ -161,25 +167,69 @@ export function createExecutionHandler<
 
       // TODO would be nice  to add the action name here, but it's not available
       const actionLabel = `${action.actionTypeId}:${action.id}`;
-      const actionsClient = await actionsPlugin.getActionsClientWithRequest(request);
-      await actionsClient.enqueueExecution({
-        id: action.id,
-        params: action.params,
-        spaceId,
-        apiKey: apiKey ?? null,
-        source: asSavedObjectExecutionSource({
-          id: alertId,
-          type: 'alert',
-        }),
-        relatedSavedObjects: [
-          {
+      if (supportsEphemeralTasks && ephemeralActionsToSchedule > 0) {
+        ephemeralActionsToSchedule--;
+        actionsClient
+          .ephemeralEnqueuedExecution({
+            id: action.id,
+            params: action.params,
+            spaceId,
+            apiKey: apiKey ?? null,
+            source: asSavedObjectExecutionSource({
+              id: alertId,
+              type: 'alert',
+            }),
+            relatedSavedObjects: [
+              {
+                id: alertId,
+                type: 'alert',
+                namespace: namespace.namespace,
+                typeId: alertType.id,
+              },
+            ],
+          })
+          .catch((err) => {
+            if (isEphemeralTaskRejectedDueToCapacityError(err)) {
+              actionsClient.enqueueExecution({
+                id: action.id,
+                params: action.params,
+                spaceId,
+                apiKey: apiKey ?? null,
+                source: asSavedObjectExecutionSource({
+                  id: alertId,
+                  type: 'alert',
+                }),
+                relatedSavedObjects: [
+                  {
+                    id: alertId,
+                    type: 'alert',
+                    namespace: namespace.namespace,
+                    typeId: alertType.id,
+                  },
+                ],
+              });
+            }
+          });
+      } else {
+        await actionsClient.enqueueExecution({
```

Comment:
nit: can we encapsulate this call in a function or move the logic to create the `EnqueueExecutionOptions` object into a function? there are a lot of options being passed in and it would be nice to only have to update one location if we need to add a new field

Reply:
Sure!